src/services/lnd/onchain-service.ts

```diff
@@ -27,26 +28,42 @@ export const OnChainService = (
     return new OnChainServiceUnavailableError(err)
   }
 
-  const getIncomingTransactions = async (
+  const listTransactions = async (
     scanDepth: number,
-  ): Promise<SubmittedTransaction[] | OnChainServiceError> => {
+  ): Promise<GetChainTransactionsResult | OnChainServiceError> => {
     try {
       const { current_block_height } = await getHeight({ lnd })
 
       // this is necessary for tests, otherwise after may be negative
       const after = Math.max(0, current_block_height - scanDepth)
 
-      const result = await getChainTransactions({
+      return getChainTransactions({
         lnd,
         after,
       })
-
-      return extractIncomingTransactions(decoder, result)
     } catch (err) {
       return new UnknownOnChainServiceError(err)
     }
   }
 
+  const listIncomingTransactions = async (
+    scanDepth: number,
+  ): Promise<SubmittedTransaction[] | OnChainServiceError> => {
```

Comment:
in my mind `SubmittedTransaction` suggested more an intent of our node to send/submit transaction (ie: outgoing transaction) versus transaction coming to us.

Reply:
:thinking: yep although this is just a refactor (this type was not introduced in this PR). what name do you suggest? (we already have `OnChainTransaction` and is part of this type https://github.com/GaloyMoney/galoy/blob/main/src/domain/bitcoin/onchain/index.types.d.ts#L25)

Reply:
`IncomingTransaction[]`?

Reply:
Hmm I think the `SubmittedTransaction` name isn't great. Its intended to signify a tx that can be in the mempool or confirmed. From looking through the code I think `IncomingTransaction` (or `IncomingOnChainTransaction`) works everywhere its being used.

Reply:
change done here https://github.com/GaloyMoney/galoy/pull/787 to avoid a bigger PR